core/source/Http.mint

```diff
@@ -152,7 +152,9 @@ module Http {
     |> Http.send()
   */
   fun jsonBody (body : Object, request : Http.Request) : Http.Request {
-    { request | body = `JSON.stringify(#{body})` }
+    { request |
+      body = `JSON.stringify(#{body})`,
+      headers = Array.push(`new Record({ value: "application/json", key: "Content-Type" })`, request.headers) }
```

Comment:
I think we should check for an already existing `Content-Type` record and skip the change if set or update it otherwise.

Reply:
Good catch @Sija. 

Should we also handle the case for when the user adds the header after calling `jsonBody` by adding a check in the `header` function?

```mint
Http.empty()
|> Http.jsonBody(encode { user = "spaceman" }))
|> Http.header("Content-type", "application/json")
```

An alternate way of handling this could be to remove duplicates before making the actual HTTP call in the `send` function.

Reply:
First, I'd fix the `Http.header` to take existing entries into account - i.e. overwriting them if found, and adding them otherwise. We could also add an additional argument to possibly skip overwriting, WDYT @gdotdesign?

Reply:
> First, I'd fix the `Http.header` to take existing entries into account - i.e. overwriting them if found, and adding them otherwise.

Overwriting existing headers sounds good :+1: 

> We could also add an additional argument to possibly skip overwriting, 

I would add a separate function instead, something like `Http.appendHeader(name, value)` which would just append and this way it wouldn't be a breaking change.

Reply:
Yes, let's add the ability to overwrite the value if key already exists in `Http.header`.

On a different note, do you think the functionality to skip overwriting if the key already exists is really needed? I mean, if the user is setting the header then they should know if a key is being set or not and if they really mean to find out, they could always check the contents of the `Http.Request` record (that they themselves have created).

If we do decide to go with it, does `Http.appendHeader(name, value)` give the right idea about internal implementation? This might seem longer but gives a more clear idea:
- `Http.appendHeaderIfNotExists(name, value)` or 
- `Http.addHeaderIfNotExists(name, value)`

Just my 2 cents, what do you guys feel?

Reply:
> I would add a separate function instead, something like `Http.appendHeader(name, value)` which would just append and this way it wouldn't be a breaking change.

Perhaps it would be a better API to add `fun hasHeader(key : String) : Bool` so the user can check whether given header *key* exists and decide by himself what's the more appropriate.

With such API design this function would need to just do (pseudo-code)

```mint
if (!Http.hasHeader("Content-Type", request)) {
  request = Http.header("Content-Type", "application/json", request)
}
request
```

Reply:
That could also be a possibility, and it's a one-liner in mint code:

```
Array.any((header : Http.Header) : Bool { header.key == key }, request.headers)
```